File: VHBoomMenuButtonExample/ControlVC.m
//
//  ControlVC.m
//  VHBoomMenuButton
//
//  Created by viktorhuang on 2017/4/14.
//  Copyright © 2017年 Nightonke. All rights reserved.
//

#import "ControlVC.h"
#import "BuilderManager.h"
#import "VHBoomMenuButton.h"

@interface ControlVC ()

@property (weak, nonatomic) IBOutlet UIButton *boomButton;
@property (weak, nonatomic) IBOutlet UIButton *reboomButton;
@property (weak, nonatomic) IBOutlet UIButton *boomImmediatelyButton;
@property (weak, nonatomic) IBOutlet UIButton *reboomImmediatelyButton;

@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb;

@end

@implementation ControlVC

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.bmb.buttonEnum = VHButtonSimpleCircle;
    self.bmb.piecePlaceEnum = VHPiecePlace_DOT_9_1;
    self.bmb.buttonPlaceEnum = VHButtonPlace_SC_9_1;
    self.bmb.boomInWholeScreen = NO;
    for (int i = 0; i < [VHBoomMenuButton pieceNumber:self.bmb.piecePlaceEnum]; i++)
    {
        [self.bmb addBuilder:[BuilderManager simpleCircleButtonBuilder]];
    }
}

- (IBAction)onButtonClicked:(UIButton *)sender
{
    if (sender == self.boomButton)
    {
        [self.bmb boom];
    }
    else if (sender == self.reboomButton)
    {
        [self.bmb reboom];
    }
    else if (sender == self.boomImmediatelyButton)
    {
        [self.bmb boomImmediately];
    }
    else if (sender == self.reboomImmediatelyButton)
    {
        [self.bmb reboomImmediately];
    }
}

@end

File: VHBoomMenuButtonSwiftDemo/ControlVC.swift
//
//  ControlVC.swift
//  VHBoomMenuButtonSwiftDemo
//
//  Created by Nightonke on 2017/5/5.
//  Copyright © 2017 Nightonke. All rights reserved.
//

import UIKit
import BoomMenuButton

class ControlVC: BoomMenuButtonVC {

    @IBOutlet weak var boomButton: UIButton!
    @IBOutlet weak var reboomButton: UIButton!
    @IBOutlet weak var boomImmediatelyButton: UIButton!
    @IBOutlet weak var reboomImmediatelyButton: UIButton!
    
    @IBOutlet weak var bmb: BoomMenuButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        bmb.buttonEnum = .simpleCircle
        bmb.piecePlaceEnum = .dot_9_1
        bmb.buttonPlaceEnum = .sc_9_1
        bmb.isBoomInWholeScreen = false
        for _ in 0..<bmb.piecePlaceEnum.pieceNumber() {
            bmb.addBuilder(BuilderManager.simpleCircleButtonBuilder())
        }
    }

    @IBAction func onButtonClicked(_ sender: UIButton) {
        if sender == boomButton {
            bmb.boom()
        } else if sender == reboomButton {
            bmb.reboom()
        } else if sender == boomImmediatelyButton {
            bmb.boomImmediately()
        } else if sender == reboomImmediatelyButton {
            bmb.reboomImmediately()
        }
    }
}
